hamilton/lifecycle/api.py

```diff
@@ -614,6 +635,42 @@ def validate_graph(
         return self.run_to_validate_graph(graph=HamiltonGraph.from_graph(graph))
 
 
+class TaskGroupingHook(BasePostTaskGroup, BasePostTaskExpand):
+    """Implement this to run something after task grouping or task expansion. This will allow you to
+    capture information about the tasks during `Parallelize`/`Collect` blocks in dynamic DAG execution."""
+
+    @override
+    @final
+    def post_task_group(self, *, run_id: str, task_ids: List[str]):
+        return self.run_after_task_grouping(run_id=run_id, task_ids=task_ids)
+
+    @override
+    @final
+    def post_task_expand(self, *, run_id: str, task_id: str, parameters: Dict[str, Any]):
```

Comment:
Idea -- maybe we allow parameters to be `None`? I think I'm overthikning this but want to guard against a future case where using force evaluates everything... If it's `Optional[...]` then we can at least guard against it. That said, it could also be an empty dict if we don't want to, and that could be specified by the hook, so let's not worry about it.